### PR TITLE
Sim2real

### PR DESCRIPTION
**Pull request by yang-zj1026**
fill up c code for attention model

### CODE REVIEW

File: gym_art/quadrotor_multi/obstacles/obstacles.py
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/obstacles/obstacles.py:1]: I removed radar representation. Please make sure your version matches with this version.

File: gym_art/quadrotor_multi/obstacles/test/speed_test.py
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/obstacles/test/speed_test.py:1]: Same as above.

File: gym_art/quadrotor_multi/obstacles/test/unit_test.py
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/obstacles/test/unit_test.py:1]: Same as above

File: gym_art/quadrotor_multi/obstacles/utils.py
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/obstacles/utils.py:1]: Same as above.

File: gym_art/quadrotor_multi/quadrotor_multi.py
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/quadrotor_multi.py:1]: Same as above

File: gym_art/quadrotor_multi/quadrotor_single.py
Comment (Zhehui-Huang) [inline gym_art/quadrotor_multi/quadrotor_single.py:1]: Same as above

File: setup.py
Comment (Zhehui-Huang) [inline setup.py:1]: Please update this file to match with my version.

File: swarm_rl/env_wrappers/quad_utils.py
Comment (Zhehui-Huang) [inline swarm_rl/env_wrappers/quad_utils.py:1]: Same as above

File: swarm_rl/env_wrappers/quadrotor_params.py
Comment (Zhehui-Huang) [inline swarm_rl/env_wrappers/quadrotor_params.py:1]: Same as above

File: swarm_rl/models/quad_multi_model.py
Comment (Zhehui-Huang) [inline swarm_rl/models/quad_multi_model.py:1]: Please clean up this file

File: swarm_rl/runs/obstacles/quads_multi_obstacles_radar.py
Comment (Zhehui-Huang) [inline swarm_rl/runs/obstacles/quads_multi_obstacles_radar.py:1]: Delete this file

File: train_brain_multi_obstacles_radar.sh
Comment (Zhehui-Huang) [inline train_brain_multi_obstacles_radar.sh:1]: Delete this file

**Review by Zhehui-Huang — CHANGES_REQUESTED** [review]

**Comment by yang-zj1026** [comment]
I updated this branch and it's up to date with paper_v1

File: swarm_rl/models/quad_multi_model.py [hunk 1]
```diff
@@ -169,11 +169,14 @@ def __init__(self, cfg, obs_space):
         # Attention Layer
         self.attention_layer = MultiHeadAttention(4, cfg.rnn_size, cfg.rnn_size, cfg.rnn_size)
 
+        # # This is for sim2real
+        # self.attention_layer = OneHeadAttention(cfg.rnn_size)
+
         # MLP Layer
-        self.feed_forward = nn.Sequential(fc_layer(3 * cfg.rnn_size, 2 * cfg.rnn_size),
+        self.feed_forward = nn.Sequential(fc_layer(3 * cfg.rnn_size, cfg.rnn_size),
```
Comment (Zhehui-Huang) [inline swarm_rl/models/quad_multi_model.py]: If we change the final feed_forward layer from 2 * cfg.rnn_size to cfg.rnn_size, will that harm the final performance? If the answer is not, can you show me the wandb report?

**Review by Zhehui-Huang — CHANGES_REQUESTED** [review]

File: swarm_rl/models/quad_multi_model.py (hunk 1 above)
Comment (yang-zj1026) [reply swarm_rl/models/quad_multi_model.py]: This is just for sim2real. In simulation it's still 2 * cfg.rnn_size

Comment (Zhehui-Huang) [reply swarm_rl/models/quad_multi_model.py]: If this is for sim2real, then, you probably need a hyperparameter to decide sim2real or just for sim.

**Review by Zhehui-Huang — CHANGES_REQUESTED** [review]

**Review by Zhehui-Huang — APPROVED** [review]

**Pull request merged**